src/Psr/Tracing/RelayOpenTelemetry.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CacheWerk\Relay\Psr\Tracing;
+
+use Relay\Relay;
+
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\TracerInterface;
+use OpenTelemetry\API\Trace\TracerProviderInterface;
+
+class RelayOpenTelemetry
+{
+    /**
+     * The Relay instance.
+     *
+     * @var \Relay\Relay
+     */
+    protected Relay $relay;
+
+    /**
+     * The OpenTelemetry tracer provider instance.
+     *
+     * @var \OpenTelemetry\API\Trace\TracerInterface
+     */
+    protected TracerInterface $tracer;
+
+    /**
+     * Creates a new `NewRelicRelay` instance.
+     *
+     * @param  \Relay\Relay  $relay
+     * @param  \OpenTelemetry\API\Trace\TracerProviderInterface  $tracerProvider
+     * @return void
+     */
+    public function __construct(Relay $relay, TracerProviderInterface $tracerProvider)
+    {
+        $this->relay = $relay;
+        $this->tracer = $tracerProvider->getTracer('Relay', (string) phpversion('relay'));
+    }
+
+    /**
+     * Executes Relay methods inside OpenTelemetry span.
+     *
+     * @param  string  $name
+     * @param  array<mixed>  $arguments
+     * @return mixed
+     */
+    public function __call(string $name, array $arguments)
+    {
+        $span = $this->tracer->spanBuilder('Relay::' . strtolower($name))
+            ->setAttribute('db.operation', strtoupper($name))
+            ->setSpanKind(SpanKind::KIND_CLIENT)
+            ->startSpan();
+
+        $result = $this->relay->{$name}(...$arguments);
```

Comment:
Any chance you can deduce whether the operation succeeded or not from the `$result`? Then record it on the span:

```php
} catch (\Exception $exc) {
    $span->setStatus("error", $exc->getMessage());
    $span->recordException($exc);
}
```

If there are no exceptions, then just `$span->setStatus("error")` will do.

Reply:
Implementing `setStatus()` is difficult, there are too many varying responses.

But I'm going to look into catching exceptions, recording them and re-throwing.